utils/build/docker/dotnet_http/parametric/Endpoints/ApmTestApi.cs

```diff
@@ -0,0 +1,325 @@
+        var parsedDictionary = JsonConvert.DeserializeObject<Dictionary<string, Object>>(headerRequestBody);
+
+        _logger?.LogInformation("StartSpan: {HeaderRequestBody}", headerRequestBody);
+
+        var creationSettings = new SpanCreationSettings
+        {
+            FinishOnClose = false,
+        };
+
+        if (parsedDictionary!.TryGetValue("http_headers", out var headersList))
+        {
+            creationSettings.Parent = _spanContextExtractor.Extract(
+                ((Newtonsoft.Json.Linq.JArray)headersList).ToObject<string[][]>()!,
+                getter: GetHeaderValues);
+        }
+
+        if (parsedDictionary!.TryGetValue("parent_id", out var parentId))
+        {
+            var longParentId = Convert.ToUInt64(parentId);
+            
+            if (creationSettings.Parent is null && longParentId > 0 )
+            {
+                var parentSpan = Spans[longParentId];
+                creationSettings.Parent = (ISpanContext)SpanContext.GetValue(parentSpan)!;
+            }
+        }
+
+        parsedDictionary.TryGetValue("name", out var name);
+        using var scope = Tracer.Instance.StartActive(operationName: name!.ToString(), creationSettings);
+        var span = scope.Span;
+
+        if (parsedDictionary.TryGetValue("service", out var service))
+        {
+            span.ServiceName = service.ToString();
+        }
+
+        if (parsedDictionary.TryGetValue("resource", out var resource))
+        {
+            span.ResourceName = resource.ToString();
+        }
+
+        if (parsedDictionary.TryGetValue("type", out var type))
+        {
+            span.Type = type.ToString();
+        }
+
+        if (parsedDictionary.TryGetValue("origin", out var origin) && !string.IsNullOrWhiteSpace(origin.ToString()))
+        {
+            var spanContext = SpanContext.GetValue(span)!;
+            Origin.SetValue(spanContext, origin);
+        }
+        
+        Spans[span.SpanId] = span;
+        
+        return JsonConvert.SerializeObject(new
+        {
+            span_id = span.SpanId.ToString(),
+            trace_id = span.TraceId.ToString(),
+        });
+    }
+    
+    private static async Task SpanSetMeta(HttpRequest request)
+    {
+        var headerBodyDictionary = await new StreamReader(request.Body).ReadToEndAsync();
+        var parsedDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(headerBodyDictionary);
+        parsedDictionary!.TryGetValue("span_id", out var id);
+        parsedDictionary.TryGetValue("key", out var key);
+        parsedDictionary.TryGetValue("value", out var value);
+
+        var span = Spans[Convert.ToUInt64(id)];
+        span.SetTag(key, value);
+    }
+
+    private static async Task SpanSetMetric(HttpRequest request)
+    {
+        var headerBodyDictionary = await new StreamReader(request.Body).ReadToEndAsync();
+        var parsedDictionary = JsonConvert.DeserializeObject<Dictionary<string, object>>(headerBodyDictionary)!;
```

Comment:
Since you do this in every method, it might be easier to break this out to a new method and call that. Whether you refactor into a new method or not, you should make sure to dispose the StreamReader once you are done with it. For example, you could do something like:

```csharp
using var streamReader = new StreamReader(request.Body);
var body = await streamReader.ReadToEndAsync();
```